src/kv/generic_serialise_wrapper.h

```diff
@@ -57,6 +57,113 @@ namespace kv
       }
     }
 
+    size_t get_serialised_size(
+      size_t serialised_public_domain_size,
+      size_t serialised_private_domain_size) const
+    {
+      size_t size =
+        sizeof(SerialisedEntryHeader) + serialised_public_domain_size;
+
+      // If no crypto util is set (unit test only), only the header and public
+      // domain are serialised
+      if (crypto_util)
+      {
+        size += crypto_util->get_header_length() + sizeof(size_t) +
+          serialised_private_domain_size;
+      }
+
+      return size;
+    }
+
+    void serialise_domains_impl(
+      const std::vector<uint8_t>& serialised_public_domain,
+      const std::vector<uint8_t>& serialised_private_domain,
+      std::span<uint8_t> buf)
```

Comment:
As discussed with @achamayou, we should also use `buf` for the serialised public and serialised private domains, to avoid allocating more memory in the enclave.